src/inverted-index.js

```diff
@@ -0,0 +1,272 @@
+        this.jsonDatabase[fileName].push(jsonFile[docObject]);
+      }
+      return true;
+    }
+    /**
+     * [getjsonDatabase function to return the saved uploads]
+     * @return {[object]} [the saved uploads]
+     */
+  getjsonDatabase() {
+    return this.jsonDatabase;
+  }
+
+  /**
+   * [createIndex Creates an index of the words in the received json file]
+   * @param  {[string]}   filePath [the key(filename) of the json value to index]
+   * @param  {Function} cb       [call back function to return the indexed file object and an html format index table]
+   * @return {[array]}            [an arrray of of two elements: the indexed file result and the html Div of the index]
+   */
+  createIndex(filePath, cb) {
+      const indexFile = this.indexFile;
+      const jsonDoc = this.jsonDatabase[filePath];
+      let concSentence = '';
+      let wordArray = [];
+      if (indexFile[filePath]) {
+        return false;
+      }
+      indexFile[filePath] = {};
+
+      jsonDoc.forEach((element, index) => {
+        concSentence = this.cleanString((`${element.title} ${element.text}`));
+        wordArray = new Set(concSentence.split(' '));
+        wordArray.forEach((word) => {
+          indexFile[filePath][word] = indexFile[filePath][word] || [];
+          indexFile[filePath][word].push(index);
+        });
+      });
+      this.indexFile = indexFile;
+      return cb(filePath, indexFile, jsonDoc);
+    }
+    /**
+     * [isValid Check if a file is a valid json object based, calls method to check structure]
+     * @param  {[type]}  fileName [the filename used to verfity is the object is already in the database]
+     * @param  {[type]}  jsonFile [the json object to be tested]
+     * @return {Boolean}          [returns true if valid else false]
+     */
+  isValid(fileName, jsonFile) {
+      if (typeof jsonFile === 'string') {
+        jsonFile = JSON.parse(jsonFile);
+      }
+
+      if (jsonFile && jsonFile.length > 0) {
+        const isValidFileStructure = this.checkFileStructure(jsonFile);
+        if (isValidFileStructure) {
+          if (!this.jsonDatabase[fileName]) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
+    /**
+     * [checkFileStructure Checks if object follows the structure as found in ./jasmine/books.json]
+     * @param  {[object]} jsonFile [json file to be tested]
+     * @return {[boolean]}          [true if valid and false if invalid]
+     */
+  checkFileStructure(jsonFile) {
+    let isValidFile = true;
+
+    jsonFile.forEach((document, documentIndex) => {
+      const isValidTitle = document.title !== undefined && document.title.length > 0 && typeof document.title === 'string';
+      const isValidText = document.text !== undefined && document.text.length > 0 && typeof document.text === 'string';
+      if (!(isValidText && isValidTitle)) {
+        isValidFile = false;
+        return false;
+      }
+    });
+    return isValidFile;
```

Comment:
**Code Quality - Nitpick** _Method Complexity_
Good job validating the file structure here but the logic is quite complex and difficult to follow.
**e.g.** `document.title !== undefined` could just be `document.title` which is a Truthy expression.
This can be refactored.